stream-chat-android-core/src/main/java/io/getstream/chat/android/models/PendingMessage.kt

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2014-2025 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.models
+
+import androidx.compose.runtime.Immutable
+
+/**
+ * Represents a message that is pending delivery or processing.
+ *
+ * @property message The [Message] object containing all the content and attributes of the message.
+ * @property metadata Additional metadata associated with this pending message, stored as key-value pairs.
+ */
+@Immutable
+public data class PendingMessage(
+    val message: Message,
+    val metadata: Map<String, String>,
+) {
+    /**
+     * Creates a new Builder instance with properties initialized from this PendingMessage.
+     *
+     * @return A Builder instance for creating a modified copy of this PendingMessage.
+     */
+    @SinceKotlin("99999.9")
+    @Suppress("NEWER_VERSION_IN_SINCE_KOTLIN")
+    public fun newBuilder(): Builder = Builder(this)
+
+    /**
+     * Builder class for constructing a [PendingMessage] instance.
+     */
+    public class Builder() {
```

Comment:
Does it make sense to have an empty builder constructor? :thinking: 
This builder was mainly though for Java User that doesn't have an easy way to call `sometthing.copy(...)`
They shouldn't build a instance from scratch.
Even if they want to build an instance from scratch, they can use the "insance constructor" with its defaults arguments instead of using the Builder class